network/p2p/listeners.go

```diff
@@ -0,0 +1,91 @@
+package p2p
+
+import (
+	"fmt"
+	"github.com/bloxapp/ssv/ibft/proto"
+	"github.com/bloxapp/ssv/network"
+	"go.uber.org/zap"
+	"time"
+)
+
+type listener struct {
+	msgCh     chan *proto.SignedMessage
+	sigCh     chan *proto.SignedMessage
+	decidedCh chan *proto.SignedMessage
+	syncCh    chan *network.SyncChanObj
+}
+
+// registerListener registers the given listener
+func (n *p2pNetwork) registerListener(ls listener) func() {
+	n.listenersLock.Lock()
+	defer n.listenersLock.Unlock()
+
+	id := fmt.Sprintf("%d:%d", len(n.listeners), time.Now().UnixNano())
+	n.listeners[id] = ls
+
+	return func() {
+		n.listenersLock.Lock()
+		defer n.listenersLock.Unlock()
+
+		delete(n.listeners, id)
+	}
+}
+
+// propagateSignedMsg takes an incoming message (from validator's topic)
+// and propagates it to the corresponding internal listeners
+func (n *p2pNetwork) propagateSignedMsg(cm *network.Message) {
+	if cm == nil || cm.SignedMessage == nil {
+		n.logger.Debug("could not propagate nil message")
+		return
+	}
+	n.logger.Debug("propagating msg to internal listeners", zap.String("type", cm.Type.String()),
```

Comment:
@amir-blox the log is printed a lot 
Can we remove it?

Reply:
Yes

Reply:
removed